### DISCUSSION TITLE
Support fields & variables in script editor

### DISCUSSION THREAD

**Discussion by steverep** [home-assistant/frontend#13771] [Automations / scripts / scenes] [answered]
One of the most useful thing about [scripts](https://www.home-assistant.io/integrations/script/) is that they support data passed as field inputs, yet these can only be added in Yaml.  I would like to see the script editor upgraded to support:
- fields
- variables
- description
- script name consistent with the new automation editor (i.e. not always a text field)

**Comment by karwosts** [comment]
Out of curiousity, if we were to add this, what do you think it should look like? AFAIK things like fields have a fairly extensive yaml settings, would it just be a big blank yaml box to type yaml into, or would we have lots of textboxes? A selector to choose a selector?

**Reply by steverep** [reply]
Funny I was just thinking about this and looking at how to maybe implement it...

For fields, my thought would just be to make it another collapsible section above the sequence/actions section, basically with the same exact format:
- A button to add a field section, each of which is collapsible
- Each field section just has a form to input the same values as in yaml
- Each field section can be removed, reordered, switched to yaml mode, etc.

I guess I don't see it as any more expensive than the forms for some triggers/conditions/actions.  I think it's much simpler actually.  WDYT?

**Reply by karwosts** [reply, marked as answer]
Took a pass at adding a fields editor in #18250. Check it out and see if it lines up with what you were thinking.
